src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,55 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+
+public class Parser {
+    /**
+     * Parses a rawCommand string into the corresponding Command object.
+     *
+     * @param rawCommand the raw command string.
+     * @return the Command object parsed from the raw command string supplied.
+     * @throws DukeException if the raw command string cannot be parsed into any Command objects.
+     */
+    public static Command parse(String rawCommand) throws DukeException {
+        String[] words = rawCommand.split(" ");
+        switch (words[0]) {
+        case "todo":
+            return new AddCommand("todo", rawCommand.substring(5));
+        case "deadline":
+        case "event":
+            StringBuilder argumentBuilder = new StringBuilder();
```

Comment:
I like that this piece of code (argumentBuilder, optionArgumentBuilder) works well for both deadlines and events 👍Perhaps you could also isolate it as a function "e.g. getDescription(.....)" on its own, that could reduce the code complexity in parse() and make it more reader-friendly.

Reply:
using StringBuilder is a good way to have efficiency instead of building a new string each time by appending strings together with +.